tests/waku_archive/test_retention_policy.nim

```diff
@@ -53,6 +54,45 @@ suite "Waku Archive - Retention policy":
 
     ## Cleanup
     (waitFor driver.close()).expect("driver to close")
+  
+  test "size retention policy - windowed message deletion":
+    ## Given
+    let
+      # in megabytes
+      sizeLimit:float = 0.05
+      excess = 123
+
+    let driver = newTestArchiveDriver()
+
+    let retentionPolicy: RetentionPolicy = SizeRetentionPolicy.init(size=sizeLimit)
+
+    ## When
+
+    var putFutures = newSeq[Future[ArchiveDriverResult[void]]]()
+    var retentionFutures = newSeq[Future[ArchiveDriverResult[void]]]()
+
+    for i in 1..excess:
+        let msg = fakeWakuMessage(payload= @[byte i], contentTopic=DefaultContentTopic, ts=Timestamp(i))
+        putFutures.add(driver.put(DefaultPubsubTopic, msg, computeDigest(msg), msg.timestamp))
+        retentionFutures.add(retentionPolicy.execute(driver))
+
+    # waitFor is used to synchronously wait for the futures to complete.
+    discard waitFor allFinished(putFutures & retentionFutures)
```

Comment:
This is great and I don't see a need to change it.
However, I think it can be reduced a little bit with:
```suggestion
    for i in 1..excess:
        let msg = fakeWakuMessage(payload= @[byte i], contentTopic=DefaultContentTopic, ts=Timestamp(i))
        waitFor driver.put(DefaultPubsubTopic, msg, computeDigest(msg), msg.timestamp)
        waitFor retentionPolicy.execute(driver)
```

Reply:
as per the previous review with Simone, the original method is more efficient?